src/views/common/Animation/AnimationStateReducer.tsx

```diff
@@ -42,6 +53,38 @@ export type AnimationStateUpdateFrameDataAction<T> = {
     replaceExistingFrames?: boolean
 }
 
+export type AnimationStateSetBaseMsPerFrameAction = {
+    type: 'SET_BASE_MS_PER_FRAME',
+    baseMsPerFrame: number
+}
+
+export type AnimationStateSetReplayRateAction = {
+    type: 'SET_REPLAY_RATE',
+    newRate: number
+}
+
+export type AnimationStateSetWindowAction = {
+    type: 'SET_WINDOW'
+    bounds?: [number, number]
+}
+
+export type AnimationStateProposeWindowAction = {
+    type: 'PROPOSE_WINDOW'
+    bounds?: [number, number]
+}
+
+export type AnimationStateCommitWindowAction = {
+    type: 'COMMIT_WINDOW'
+}
+
+export type AnimationStateReleaseWindowAction = {
+    type: 'RELEASE_WINDOW'
+}
+
```

Comment:
The overall animation state tracks the (proposed or realized) visible window. This is technically necessary (since we can't just persist it as pixels--the data flow doesn't work out right) but also means that you can bounce back and forth from the restricted view to the full-length view without losing the window selection.